// casdoor-nodejs-sdk/lib/cjs/adapter.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdapterSDK = void 0;
class AdapterSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getAdapters() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-adapters', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getAdapter(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-adapter', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyAdapter(method, adapter) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        adapter.owner = this.config.orgName;
        return (await this.request.post(url, adapter, {
            params: {
                id: `${adapter.owner}/${adapter.name}`,
            },
        }));
    }
    async addAdapter(adapter) {
        return this.modifyAdapter('add-adapter', adapter);
    }
    async updateAdapter(adapter) {
        return this.modifyAdapter('update-adapter', adapter);
    }
    async deleteAdapter(adapter) {
        return this.modifyAdapter('delete-adapter', adapter);
    }
}
exports.AdapterSDK = AdapterSDK;
