// casdoor-nodejs-sdk/lib/cjs/application.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSDK = void 0;
class ApplicationSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getApplications() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-applications', {
            params: {
                owner: 'admin',
            },
        }));
    }
    async getApplication(name) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-application', {
            params: {
                id: `admin/${name}`,
            },
        }));
    }
    async modifyApplication(method, application) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        application.owner = 'admin';
        return (await this.request.post(url, application, {
            params: {
                id: `${application.owner}/${application.name}`,
            },
        }));
    }
    async addApplication(application) {
        return this.modifyApplication('add-application', application);
    }
    async updateApplication(application) {
        return this.modifyApplication('update-application', application);
    }
    async deleteApplication(application) {
        return this.modifyApplication('delete-application', application);
    }
}
exports.ApplicationSDK = ApplicationSDK;
