// casdoor-nodejs-sdk/lib/cjs/enforcer.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnforcerSDK = void 0;
class EnforcerSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getEnforcers() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-enforcers', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getEnforcer(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-enforcer', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyEnforcer(method, enforcer) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        enforcer.owner = this.config.orgName;
        return (await this.request.post(url, enforcer, {
            params: {
                id: `${enforcer.owner}/${enforcer.name}`,
            },
        }));
    }
    async addEnforcer(enforcer) {
        return this.modifyEnforcer('add-enforcer', enforcer);
    }
    async updateEnforcer(enforcer) {
        return this.modifyEnforcer('update-enforcer', enforcer);
    }
    async deleteEnforcer(enforcer) {
        return this.modifyEnforcer('delete-enforcer', enforcer);
    }
}
exports.EnforcerSDK = EnforcerSDK;
