// casdoor-nodejs-sdk/lib/cjs/payment.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentSDK = void 0;
class PaymentSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getPayments() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-payments', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getPayment(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-payment', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyPayment(method, payment) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        payment.owner = this.config.orgName;
        return (await this.request.post(url, payment, {
            params: {
                id: `${payment.owner}/${payment.name}`,
            },
        }));
    }
    async addPayment(payment) {
        return this.modifyPayment('add-payment', payment);
    }
    async updatePayment(payment) {
        return this.modifyPayment('update-payment', payment);
    }
    async deletePayment(payment) {
        return this.modifyPayment('delete-payment', payment);
    }
}
exports.PaymentSDK = PaymentSDK;
