// casdoor-nodejs-sdk/lib/cjs/plan.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlanSDK = void 0;
class PlanSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getPlans() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-plans', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getPlan(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-plan', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyPlan(method, plan) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        plan.owner = this.config.orgName;
        return (await this.request.post(url, plan, {
            params: {
                id: `${plan.owner}/${plan.name}`,
            },
        }));
    }
    async addPlan(plan) {
        return this.modifyPlan('add-plan', plan);
    }
    async updatePlan(plan) {
        return this.modifyPlan('update-plan', plan);
    }
    async deletePlan(plan) {
        return this.modifyPlan('delete-plan', plan);
    }
}
exports.PlanSDK = PlanSDK;
