// casdoor-nodejs-sdk/lib/cjs/pricing.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingSDK = void 0;
class PricingSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getPricings() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-pricings', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getPricing(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-pricing', {
            params: {
                id: `${this.config.orgName}/${id}`,
                pageSize: 1000,
            },
        }));
    }
    async modifyPricing(method, pricing) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        pricing.owner = this.config.orgName;
        return (await this.request.post(url, pricing, {
            params: {
                id: `${pricing.owner}/${pricing.name}`,
            },
        }));
    }
    async addPricing(pricing) {
        return this.modifyPricing('add-pricing', pricing);
    }
    async updatePricing(pricing) {
        return this.modifyPricing('update-pricing', pricing);
    }
    async deletePricing(pricing) {
        return this.modifyPricing('delete-pricing', pricing);
    }
}
exports.PricingSDK = PricingSDK;
