// casdoor-nodejs-sdk/lib/cjs/provider.js
"use strict";
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderSDK = void 0;
class ProviderSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getProviders() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-providers', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getProvider(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-provider', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyProvider(method, provider) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        provider.owner = this.config.orgName;
        return (await this.request.post(url, provider, {
            params: {
                id: `${provider.owner}/${provider.name}`,
            },
        }));
    }
    async addProvider(provider) {
        return this.modifyProvider('add-provider', provider);
    }
    async updateProvider(provider) {
        return this.modifyProvider('update-provider', provider);
    }
    async deleteProvider(provider) {
        return this.modifyProvider('delete-provider', provider);
    }
}
exports.ProviderSDK = ProviderSDK;
