// casdoor-nodejs-sdk/lib/esm/organization.js
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
export class OrganizationSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getOrganizations() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-organizations', {
            params: {
                owner: 'admin',
            },
        }));
    }
    async getOrganization(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-organization', {
            params: {
                id: `admin/${id}`,
            },
        }));
    }
    async modifyOrganization(method, organization) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        organization.owner = 'admin';
        return (await this.request.post(url, organization, {
            params: {
                id: `${organization.owner}/${organization.name}`,
            },
        }));
    }
    async addOrganization(organization) {
        return this.modifyOrganization('add-organization', organization);
    }
    async updateOrganization(organization) {
        return this.modifyOrganization('update-organization', organization);
    }
    async deleteOrganization(organization) {
        return this.modifyOrganization('delete-organization', organization);
    }
}
