// casdoor-nodejs-sdk/lib/esm/permission.js
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
export class PermissionSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getPermissions() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-permissions', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getPermission(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-permission', {
            params: {
                id: `${this.config.orgName}/${id}`,
            },
        }));
    }
    async modifyPermission(method, permission) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        permission.owner = this.config.orgName;
        return (await this.request.post(url, permission, {
            params: {
                id: `${permission.owner}/${permission.name}`,
            },
        }));
    }
    async addPermission(permission) {
        return this.modifyPermission('add-permission', permission);
    }
    async updatePermission(permission) {
        return this.modifyPermission('update-permission', permission);
    }
    async deletePermission(permission) {
        return this.modifyPermission('delete-permission', permission);
    }
}
