// casdoor-nodejs-sdk/lib/esm/session.js
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
export class SessionSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getSessions() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-sessions', {
            params: {
                owner: this.config.orgName,
            },
        }));
    }
    async getSession(name, application) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-session', {
            params: {
                sessionPkId: `${this.config.orgName}/${name}/${application}`,
            },
        }));
    }
    async modifySession(method, session) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        session.owner = this.config.orgName;
        return (await this.request.post(url, session, {
            params: {
                id: `${session.owner}/${session.name}`,
            },
        }));
    }
    async addSession(session) {
        return this.modifySession('add-session', session);
    }
    async updateSession(session) {
        return this.modifySession('update-session', session);
    }
    async deleteSession(session) {
        return this.modifySession('delete-session', session);
    }
}
