// casdoor-nodejs-sdk/lib/esm/subscription.js
// Copyright 2021 The Casdoor Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
export class SubscriptionSDK {
    constructor(config, request) {
        this.config = config;
        this.request = request;
    }
    async getSubscriptions() {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-subscriptions', {
            params: {
                owner: this.config.orgName,
                pageSize: 1000,
            },
        }));
    }
    async getSubscription(id) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        return (await this.request.get('/get-subscription', {
            params: {
                id: `${this.config.orgName}/${id}`,
                pageSize: 1000,
            },
        }));
    }
    async modifySubscription(method, subscription) {
        if (!this.request) {
            throw new Error('request init failed');
        }
        const url = `/${method}`;
        subscription.owner = this.config.orgName;
        return (await this.request.post(url, subscription, {
            params: {
                id: `${subscription.owner}/${subscription.name}`,
            },
        }));
    }
    async addSubscription(subscription) {
        return this.modifySubscription('add-subscription', subscription);
    }
    async updateSubscription(subscription) {
        return this.modifySubscription('update-subscription', subscription);
    }
    async deleteSubscription(subscription) {
        return this.modifySubscription('delete-subscription', subscription);
    }
}
